prow/cmd/webhook-server/main.go

```diff
@@ -212,29 +219,25 @@ func handleSecrets(client ClientInterface, ctx context.Context, clientoptions cl
 		if err != nil {
 			return "", "", fmt.Errorf("unable to create ca certificate %v", err)
 		}
-		if err = createValidatingWebhookConfig(ctx, caPem, cl); err != nil {
-			return "", "", fmt.Errorf("unable to generate ValidationWebhookConfig %v", err)
-		}
 	} else {
 		err = json.Unmarshal(data, &secretsMap)
 		if err != nil {
 			return "", "", fmt.Errorf("error marshalling CA cert secret data: %v", err)
 		}
 		cert = secretsMap[certFile]
 		privKey = secretsMap[privKeyFile]
-	}
-
-	if err := isCertValid(cert); err != nil {
-		logrus.WithError(err).Info("Certificate is not valid, will replace.")
-		cert, privKey, caPem, err = updateSecret(client, ctx, clientoptions)
-		if err != nil {
-			return "", "", fmt.Errorf("unable to update secret %v", err)
-		}
-		if err := patchValidatingWebhookConfig(ctx, caPem, cl); err != nil {
-			return "", "", fmt.Errorf("unable to generate ValidationWebhookConfig %v", err)
+		if err := isCertValid(cert); err != nil {
+			logrus.WithError(err).Info("Certificate is not valid, will replace.")
+			cert, privKey, caPem, err = updateSecret(client, ctx, clientoptions)
+			if err != nil {
+				return "", "", fmt.Errorf("unable to update secret %v", err)
+			}
 		}
 	}
-
+	err = reconcileWebhooks(ctx, caPem, cl)
+	if err != nil {
```

Comment:
combine these two lines, it's a general practice in this case for reduced variable scope